rep-0144.rst

```diff
@@ -0,0 +1,129 @@
+* a package name should not contain ``ros`` as it is redundant. Exceptions include 
+  core packages and ROS bindings of an upstream library (e.g. ``moveit_ros``)
+* one of ROS's goals is to develop a canonical set of tools for making robots do
+  interesting things. The package name should describe what the package does, not where
+  it came from
+* to check whether a name is taken, consult [3]_. If you'd like your
+  repository included in that list, see the tutorial at [4]_
+
+Naming Rules
+------------
+
+The following rules define the different parts of the package name. The overall 
+idea is to prepend a name with words that distinguish it from similar 
+functional implementation (e.g.: ``lab``, ``robot``) but in order of importance 
+(e.g.: ``python_robot_lab``). Similarly, words that specialize this 
+functionality are appended (e.g. ``msgs``, ``config`` ...).
+
+The rules to add those words should be followed in order. For prefixes:
+
+* if a package is specialized by an entity (lab, company, ...), prepend the 
+  name of the entity
+* if a package is specialized for a software project, prepend its name
+* if a package is specialized for a hardware piece, prepend its name
+* if a package is specialized for a robot, prepend its name
+
+For suffixes:
+
+* if a package is a driver, append ``driver``
+* if a package contains any of a ROS message/service/action, append ``msgs``
+* if a package is a plugin for a library, append ``library-name_plugins``, e.g. 
+  ``pr2_gazebo_plugins``
+* if a package is meant for test only, append ``test``
+
+Special Suffixes:
+
+* a development only package, containing anything used to compile against 
+  (headers, CMake files ...) should have its name end in ``dev``
+* a package built in debug mode should have its name end in ``dbg``
+* a meta package for a robot should be named ``name-of-the-robot_robot``, e.g. 
+  ``pr2_robot``
+* a package containing the URDF and meshes of a robot should be named 
+  ``name-of-the-robot_description``, e.g ``pr2_willow_garage_description``
+
+Special Cases
+-------------
+
+* a Python package should have its name starting with ``python`` to follow the 
+  Debian usage
+* a package containing only a set of launch files should end with ``launch``
+* a package containing one or more tutorials only should end with ``tutorials``
+* a package containing one or more demos only should end with ``demos``
+* a package containing one or more tests only should end with ``tests``
+* third party libraries that are patched / integrated into ROS should not be named
+  like their rosdep key as it creates a conflict across Ubuntu versions. If it is
+  not specialized, name it generically ``name-of-library_ros``
+
+Package Version
+===============
+
+The versioning should follow the Debian convention as detailed in [5]_ : it may
+contain only alphanumerics and the characters ``. + - : ~``
+(full stop, plus, hyphen, colon, tilde) and should start with a digit.
```

Comment:
I think this is actually not strict enough, and our infrastructure can almost certainly not handle anything outside of `major.minor.patch`. I think this is a reasonable scheme for version because other tools also do not support more extreme cases, e.g. CMake only supports `major.minor.patch.tweak` and the Semantic Versioning scheme only allows `major.minor.patch` plus a few trailing modifiers for pre-releases (which we don't really do with our current infrastructure).

I would be happy to discuss expanding our support of different version number schemes, but I think it would be hard to justify the full complexity allowed by the Debian policy.